packages/aws-cdk-lib/aws-stepfunctions/test/distributed-map.test.ts

```diff
@@ -631,6 +634,9 @@ describe('Distributed Map State', () => {
           ToleratedFailurePercentagePath: '$.toleratedFailurePercentage',
           ToleratedFailureCountPath: '$.toleratedFailureCount',
           ItemBatcher: {
+            BatchInput: {
+              'TestBatchInput.$': '$.testBatchInput',
```

Comment:
:+1: to this, a brief comment here explaining why we need the `.$` and what it signifies (likely with a link to the relevant docs) would be very helpful

Reply:
Can do! It might be a little while for a proper response. I'm away for the next week, but will try to get to it when I get back. 

The long and short of it is that stepfunctions will parse keys that end in .$ as json paths and will look up the value from previous states using the value of this key rather than just passing the value to the next state. 

This is done in almost all other state configuration so that values can be dynamically included from previous states, but this functionality was missed here requiring the user to manually specify '.$'. This is confusing and it isn't clear that the json paths aren't automatically rendered by the cdk construct parsing. 

See documentation here: https://docs.aws.amazon.com/step-functions/latest/dg/connect-parameters.html#connect-parameters-path

And here: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions.JsonPath.html